### PR TITLE
MNTOR-2955: onRemoveEmail returns proper errors

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-2955

# Description
n/a

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts [hunk 1]
```diff
@@ -135,18 +135,35 @@ export async function onRemoveEmail(email: EmailRow) {
...
   if (email.subscriber_id !== subscriber?.id) {
     logger.error(
-      `Subscriber [${subscriber?.id}] tried to delete email [${email.id}], which belongs to another subscriber.`,
+      `User [${subscriber?.id}] tried to delete email [${email.id}], which belongs to another user.`,
     );
-    return;
+    return {
+      success: false,
+      error: "delete-email-without-permission",
+      errorMessage: `User [${subscriber?.id}] tried to delete email [${email.id}], which belongs to another user.`,
```
Comment (rhelmer) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:154]: This is returned to the caller right? I think the error message shouldn't include the ID details.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts [hunk 2]
```diff
@@ -135,18 +135,35 @@ export async function onRemoveEmail(email: EmailRow) {
...
+
+  try {
+    await removeOneSecondaryEmail(email.id);
+    await deleteResolutionsWithEmail(email.subscriber_id, email.email);
+    revalidatePath("/user/settings/");
+  } catch (e) {
+    return {
+      success: false,
+      error: "delete-email-error",
+      errorMessage: (e as Error).message,
```
Comment (rhelmer) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:166]: Similar here, we don't know what might be in the `Error` message so should not be sent to the caller (unless I'm wrong and that's not what this return is doing).

**Review by rhelmer — CHANGES_REQUESTED** [review]

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts [hunk 3]
```diff
@@ -138,7 +139,7 @@ export async function onRemoveEmail(email: EmailRow) {
     return {
       success: false,
       error: "delete-email-without-active-session",
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:141]: What is this `error` ID lookin' thing?
I assumed it was an FTL id and `errorMessage` was some fallback text, but not seeing other instances of "delete-email-without-permission" in the codebase.

```sh
[MNTOR-2955] $ git grep -in 'delete-email-without-permission' | cat
src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:154:      error: "delete-email-without-permission",
```

Also tried looking up by the `errorMessage` but didn't find matches in .ftl files and am wondering if this is possibly just hardcoded English in the dashboard:
```sh
[MNTOR-2955] $ git grep -in -B1 'without an active session' | cat
src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts-136-    logger.error(
src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:137:      `Tried to delete email [${email.id}] without an active session.`,
--
src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts-141-      error: "delete-email-without-active-session",
src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:142:      errorMessage: `User tried to delete email without an active session.`,
```

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts [hunk 4]
```diff
@@ -138,7 +139,7 @@ export async function onRemoveEmail(email: EmailRow) {
...
-      errorMessage: `Tried to delete email [${email.id}] without an active session.`,
+      errorMessage: `User tried to delete email without an active session.`,
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:142]: I guess related to above pondering&hellip; Should this string be in an FTL file similar to L167 below?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts [hunk 5]
```diff
@@ -151,7 +152,7 @@ export async function onRemoveEmail(email: EmailRow) {
     return {
       success: false,
       error: "delete-email-without-permission",
-      errorMessage: `User [${subscriber?.id}] tried to delete email [${email.id}], which belongs to another user.`,
+      errorMessage: `User tried to delete an email that belongs to another user.`,
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:155]: Same Q here re: hardcoded vs `l10n.getString("...")` like L167 below.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts (hunk 3 above)
Comment (rhelmer) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:141]: I don't think we should translate error messages, they aren't shown directly to customers - they are for consumers of the API. We just need some kind of identifier that will show up in our logs.

**Review by flodolo — APPROVED** [review]

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts [hunk 6]
```diff
@@ -130,23 +130,41 @@ export async function onAddEmail(
...
+      success: false,
+      error: "delete-email-without-permission",
+      errorMessage: `User tried to delete an email that belongs to another user.`,
+    };
+  }
+
+  try {
+    await removeOneSecondaryEmail(email.id);
+    await deleteResolutionsWithEmail(email.subscriber_id, email.email);
+    revalidatePath("/user/settings/");
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/actions.ts:162]: Should this do something like `return { success: true };` for consistency?

**Pull request merged**